src/App.tsx

```diff
@@ -15,36 +17,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+function getPreparedGoods(
+  goods: string[],
+  reverse: boolean,
+  { sortField }: SortParams,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Name:
+          return good1.localeCompare(good2);
+
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, isReverse, {
+    sortField,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortField(SortType.Name)}
+          type="button"
+          className={`button is-info ${sortField !== SortType.Name && 'is-light'}`}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortField(SortType.Length)}
+          type="button"
+          className={`button is-success ${sortField !== SortType.Length && 'is-light'}`}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() => (isReverse ? setIsReverse(false) : setIsReverse(true))}
```

Comment:
```suggestion
          onClick={() => (setIsReverse((prevReverse)=>!prevReverse)}
```
simplify and pls move the function inside `onClick` handler to a separate const and use as a callback here

**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**

Reply:
please, have a look if I fixed correctly